src/poc/miner_poc_denylist.erl

```diff
@@ -0,0 +1,184 @@
+                                                                    (_, _, _, Acc) ->
+                                                                            Acc
+                                                                    end, none, {"zipfile", ZipBin}) of
+                                                none ->
+                                                    lager:notice("zip file for denylist release did not contain \"denyfile\" file"),
+                                                    {noreply, schedule_check(State)};
+                                                <<Version:8/integer, SignatureLen:16/integer-unsigned-little, Signature:SignatureLen/binary, Rest/binary>> = Bin when Version == 1 ->
+                                                    %% check signature is still valid against our key
+                                                    case libp2p_crypto:verify(Rest, Signature, libp2p_crypto:b58_to_pubkey(Key)) of
+                                                        true ->
+                                                            <<Serial:32/integer-unsigned-little, FilterBin/binary>> = Rest,
+                                                            case xorf:from_bin(FilterBin) of
+                                                                {ok, Filter} ->
+                                                                    BaseDir = application:get_env(blockchain, base_dir, "data"),
+                                                                    DenyFile = filename:join([BaseDir, "denylist", "latest"]),
+                                                                    TmpDenyFile = DenyFile ++ "-tmp",
+                                                                    case file:write_file(TmpDenyFile, Bin) of
+                                                                        ok ->
+                                                                            case file:rename(TmpDenyFile, DenyFile) of
+                                                                                ok ->
+                                                                                    ok;
+                                                                                {error, RenameReason} ->
+                                                                                    lager:notice("failed to rename ~p to ~p: ~p", [TmpDenyFile, DenyFile, RenameReason])
+                                                                            end;
+                                                                        {error, WriteReason} ->
+                                                                            lager:notice("failed to write denyfile ~p to disk ~p", [TmpDenyFile, WriteReason])
+                                                                    end,
+                                                                    ok = persistent_term:put(?MODULE, Filter),
```

Comment:
seems like a little wrangling with that tmpfile on the disk-loaded file and you could combine this with the read-file/load persistent term functionality in the `init/1`

Reply:
This is tricky because the error paths differ, so I didn't do it.